pkg/proxy/proxy_test.go

```diff
@@ -219,6 +255,8 @@ func (s *TestProxySuite) checkPlainHTTPErrors(fakeApp *fake.ProxyFakeApp) {
 
 		s.Run("internal error if accessing incorrect url", func() {
 			// given
+			fakeApp.Err = errors.New("some-error")
+			defer func() { fakeApp.Err = nil }()
```

Comment:
We need it because `fakeApp` is shared across multiple test cases. The test cases we are adding after this one are impacted by its changes to `fakeApp`, so we have to cleanup before running the next one.
Without this line the tests we are adding in this PR inherit the change to `fakeApp.Err` and, as a result, `fakeApp` will return the error `some-error` even in the new tests. 

I added this simple cleanup so that the default `fakeApp` behavior is restored to default. Basically, now the test `internal error if get accesses returns an error` undoes the changes it has done to the `fakeApp` before exiting, leaving `fakeApp` in the default state.